apps/functions-subscription/src/adapters/azure/functions/middleware.ts

```diff
@@ -33,26 +33,34 @@ export const parsePathParameter =
       E.mapLeft(({ message }) => new H.HttpBadRequestError(message)),
     );
 
+type AllowedGroup = 'ApiTrialUser' | 'ApiTrialManager';
 /**
  * Verifies the presence of the `x-user-groups` header and checks if it includes
- * the specified group. If the `x-user-groups` header is missing,
+ * at least one of the specified groups. If the `x-user-groups` header is missing,
  * `verifyUserGroup` permits the request as if the group were included in the
  * header.
  */
 export const verifyUserGroup =
-  (group: 'ApiTrialManager') => (req: H.HttpRequest) =>
+  (allowedGroups: readonly AllowedGroup[]) => (req: H.HttpRequest) =>
     pipe(
       req.headers['x-user-groups'],
       E.fromNullable(void 0),
       E.foldW(
-        // if x-user-groups does not exists behave like it were included
+        // if x-user-groups does not exist behave like it were included
         E.right,
         // if exists then verify if it is included
         flow(
           H.parse(t.string, `Invalid format of 'x-user-groups' header`),
+          E.map((stringGroups) => stringGroups.split(',')),
           E.filterOrElseW(
-            (stringGroups) => stringGroups.split(',').includes(group),
-            () => new H.HttpForbiddenError(`Missing required group: ${group}`),
+            (headerGroups) =>
+              headerGroups.some((group) =>
+                allowedGroups.toString().includes(group),
```

Comment:
Why do we convert here using `toString()`? I guess it to avoid casting.

If `allowedGroups = ['ApiTrialUser', 'ApiTrialManager'] and group = 'TrialUser', allowedGroups.toString().includes('TrialUser')` would return `true`.

```suggestion
                AllowedGroup.is(group),
```

with some `AllowedGroup` typeguard.